### ISSUE TITLE
Issue with GP prior size 

### ISSUE THREAD

**Issue by chopley** [pymc-devs/pymc#6278]
I am trying to reproduce the content in this blogpost: https://www.pymc-labs.io/blog-posts/spatial-gaussian-process-01/

PyMC: v4.0.0b5
Python: 3.9.10
Using Conda

I seem to have an issue with this:
https://github.com/pymc-devs/pymc/blob/f80e45c6dc25737c26234b190cccdbaf89f2b5ac/pymc/gp/gp.py#L153

```
X = np.array(list(county_lonlat.values()))
with pm.Model(
    coords={
        "county": counties,
        "household": df.index,
        "feature": ["longitude", "latitude"],
    }
) as m:
    _county_idx = pm.ConstantData("county_idx", county_idx, dims="household")
    _X = pm.ConstantData("county_X", X, dims=("county", "feature"))
```

```
with m:
    state_mu = pm.Normal("state_mu", 0, 3)
    state_scale = pm.HalfNormal("state_scale", 0.7)
```
```
with m:
    ls = pm.Gamma("ls", mu=200, sigma=50)
    latent = pm.gp.Latent(cov_func=Matern32Chordal(2, ls),)
    county_eps = latent.prior("county_eps", _X, dims="county", jitter=1e-7)
```

Error message
```
/opt/conda/envs/pymc-dev-v4.0.0b5/lib/python3.9/site-packages/pymc/gp/cov.py:95: UserWarning: Only 2 column(s) out of Subtensor{int64}.0 are being used to compute the covariance function. If this is not intended, increase 'input_dim' parameter to the number of columns to use. Ignore otherwise.
  warnings.warn(
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File /opt/conda/envs/pymc-dev-v4.0.0b5/lib/python3.9/site-packages/pymc/gp/util.py:108, in infer_size(X, n_points)
    107 try:
--> 108     n_points = int(X.shape[0])
    109 except TypeError:

TypeError: __trunc__ returned non-Integral (type TensorVariable)

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
Input In [24], in <module>
      2 ls = pm.Gamma("ls", mu=200, sigma=50)
      3 latent = pm.gp.Latent(cov_func=Matern32Chordal(2, ls),)
----> 4 county_eps = latent.prior("county_eps", _X, dims="county", jitter=1e-7)

File /opt/conda/envs/pymc-dev-v4.0.0b5/lib/python3.9/site-packages/pymc/gp/gp.py:169, in Latent.prior(self, name, X, reparameterize, jitter, **kwargs)
    141 def prior(self, name, X, reparameterize=True, jitter=JITTER_DEFAULT, **kwargs):
    142     R"""
    143     Returns the GP prior distribution evaluated over the input
    144     locations `X`.
   (...)
    166         Extra keyword arguments that are passed to distribution constructor.
    167     """
--> 169     f = self._build_prior(name, X, reparameterize, jitter, **kwargs)
    170     self.X = X
    171     self.f = f

File /opt/conda/envs/pymc-dev-v4.0.0b5/lib/python3.9/site-packages/pymc/gp/gp.py:134, in Latent._build_prior(self, name, X, reparameterize, jitter, **kwargs)
    132 cov = stabilize(self.cov_func(X), jitter)
    133 if reparameterize:
--> 134     size = infer_size(X, kwargs.pop("size", None))
    135     v = pm.Normal(name + "_rotated_", mu=0.0, sigma=1.0, size=size, **kwargs)
    136     f = pm.Deterministic(name, mu + cholesky(cov).dot(v))

File /opt/conda/envs/pymc-dev-v4.0.0b5/lib/python3.9/site-packages/pymc/gp/util.py:110, in infer_size(X, n_points)
    108         n_points = int(X.shape[0])
    109     except TypeError:
--> 110         raise TypeError("Cannot infer 'shape', provide as an argument")
    111 return n_points

TypeError: Cannot infer 'shape', provide as an argument
```

**Comment by chopley** [comment]
I think it is just the old version of PyMC actually- will check.

**Comment by michaelosthege** [comment]
Yes, I fixed this bug in 9ca57bfdd36c7c40f097c59f2f461767ad6fb465
